src/transforms/filter.rs

```diff
@@ -81,7 +87,16 @@ impl FunctionTransform for Filter {
         if self.condition.check(&event) {
             output.push(event);
         } else {
-            emit!(&FilterEventDiscarded);
+            // TODO without consulting the passage of time it's possible that a
+            // slow-poke could never see this metric.
```

Comment:
An alternative here would be to track this on the `TransformOutputsBuf` or similar. Those are persistent for whole batches of input events, so you could do an emit at the same time we send the transformed outputs downstream.

Reply:
Ah, that's a good idea.

Reply:
I ultimately went with a simple time trigger.